components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/IdentityProviderManagementServiceTest.java

```diff
@@ -0,0 +1,1097 @@
+
+        ClaimMetadataManagementServiceImpl claimMetadataManagementService =
+                mock(ClaimMetadataManagementServiceImpl.class);
+        IdpMgtServiceComponentHolder.getInstance().setClaimMetadataManagementService(claimMetadataManagementService);
+
+        LocalClaim localClaim1 = new LocalClaim("http://wso2.org/claims/test1");
+        List<LocalClaim> claimList = new ArrayList<>();
+        claimList.add(localClaim1);
+        when(claimMetadataManagementService.getLocalClaims(anyString())).thenReturn(claimList);
+
+        String[] allLocalClaimUris = identityProviderManagementService.getAllLocalClaimUris();
+        Assert.assertEquals(allLocalClaimUris.length, 1);
+        Assert.assertEquals(allLocalClaimUris[0], "http://wso2.org/claims/test1");
+    }
+
+    @Test
+    public void testGetAllLocalClaimUrisException() throws Exception {
+
+        IdentityProviderManagementService identityProviderManagementService = new IdentityProviderManagementService();
+
+        ClaimMetadataManagementServiceImpl claimMetadataManagementService =
+                mock(ClaimMetadataManagementServiceImpl.class);
+        IdpMgtServiceComponentHolder.getInstance().setClaimMetadataManagementService(claimMetadataManagementService);
+        when(claimMetadataManagementService.getLocalClaims(anyString())).thenThrow(ClaimMetadataException.class);
+
+        assertThrows(IdentityProviderManagementException.class, () ->
+                identityProviderManagementService.getAllLocalClaimUris());
+    }
+
+    @Test
+    public void testGetAllFederatedAuthenticators() throws Exception {
+
+        IdentityProviderManagementService identityProviderManagementService = new IdentityProviderManagementService();
+
+        FederatedAuthenticatorConfig[] allFederatedAuthenticators =
+                identityProviderManagementService.getAllFederatedAuthenticators();
+        Assert.assertEquals(allFederatedAuthenticators.length, 0);
+
+        FederatedAuthenticatorConfig fac1 = mock(FederatedAuthenticatorConfig.class);
+        fac1.setDisplayName("DisplayName1");
+        fac1.setName("Name1");
+        fac1.setEnabled(true);
+        FederatedAuthenticatorConfig fac2 = mock(FederatedAuthenticatorConfig.class);
+        fac2.setDisplayName("DisplayName2");
+        fac2.setName("Name2");
+        fac2.setEnabled(true);
+
+        ApplicationAuthenticatorService.getInstance().addFederatedAuthenticator(fac1);
+        allFederatedAuthenticators = identityProviderManagementService.getAllFederatedAuthenticators();
+        Assert.assertEquals(allFederatedAuthenticators.length, 1);
+
+        ApplicationAuthenticatorService.getInstance().addFederatedAuthenticator(fac2);
+        allFederatedAuthenticators = identityProviderManagementService.getAllFederatedAuthenticators();
+        Assert.assertEquals(allFederatedAuthenticators.length, 2);
+
+        // clear after the test.
```

Comment:
Follow sentence case in the comments.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3493/commits/ef44da16e93a4a038f814603c42eadfb3a300f15